test/decl/protocol/req/unsatisfiable.swift

```diff
@@ -58,3 +58,25 @@ protocol P4 {
 protocol P5 {
   associatedtype Y where Y : S // expected-error {{type 'Self.Y' constrained to non-protocol, non-class type 'S'}}
 }
+
+protocol P6 {
+  associatedtype T
+  associatedtype U
+
+  func foo() where T == U
+  // expected-error@-1 {{instance method requirement 'foo()' cannot add constraint 'Self.T == Self.U' on 'Self'}}
+  // expected-note@-2 {{protocol requires function 'foo()' with type '() -> ()'; do you want to add a stub?}}
+}
+
+struct S2 : P6 {
+  // expected-error@-1 {{type 'S2' does not conform to protocol 'P6'}}
+  typealias T = Int
+  typealias U = String
+
+  func foo() {}
+  // expected-note@-1 {{candidate has non-matching type '() -> ()'}}
+
+  // FIXME: This error is bogus and should be omitted on account of the protocol requirement itself
+  // being invalid.
```

Comment:
Thanks! I will see what I can do about this.

Reply:
I think we're still doing getOpenedTypeOfMemberReference(), and matching up the requirement's opened type with the witness type in the solver, etc, which fails because the Self constraint isn't satisfied.

Fixing this would actually be nice because we can produce a misleading diagnostic even if the protocol requirement is valid as written:

```
protocol P {
  func foo<T>(_: T)
}

struct S : P {
  func foo<T : P>(_: T) {}
}


q.swift:5:8: error: type 'S' does not conform to protocol 'P'
struct S : P {
       ^
q.swift:6:8: note: candidate has non-matching type '<T> (T) -> ()'
  func foo<T : P>(_: T) {}
       ^
q.swift:2:8: note: protocol requires function 'foo' with type '<T> (T) -> ()'; do you want to add a stub?
  func foo<T>(_: T)
       ^
```

The problem here isn't that the types are mismatched, but that the witness has a generic constraint not satisfied by the requirement.